### ISSUE TITLE
WDI5 does not work with chromium or without chromedriver

### ISSUE THREAD

**Issue by QA-ler** [ui5-community/wdi5#428]
**Describe the bug**
WDI5 is unable to receive the webelement in following conditions:
1. Remove chromedriver from services
2. Use chromedriver service with chromium installed using 
browserName: "chromium" in wdio conf
and set Environment Variables see [chromedirver documentation](https://www.npmjs.com/package/chromedriver)
DETECT_CHROMEDRIVER_VERSION=true
INCLUDE_CHROMIUM=true
3. As in step 2, but with Dockerfile and 
ENV DETECT_CHROMEDRIVER_VERSION true
ENV INCLUDE_CHROMIUM true
We use 
`RUN apt-get update && \
    apt-get install -y chromium`
   to install chromium in docker

**To Reproduce**
1. Checkout https://github.com/ui5-community/wdi5-workshop
2. Start Webservice in Folder \ui5con2022\ui5-js-app 
` npm run serve`
3. In \ui5con2022\ui5-js-app\wdio.conf.js comment out the Chromedriver service 
![image](https://user-images.githubusercontent.com/121550022/219613076-20fe6626-5204-4d5d-9eb3-ceb2f879ef08.png)
4. In new Terminal in Folder ui5con2022\ui5-js-app run
 `npm run wdi5`
5. Error in console occurs
![image](https://user-images.githubusercontent.com/121550022/219617649-ea20f8fb-5526-42c4-886a-342e22ec05a0.png)

Error seems to appear in this line:
![image](https://user-images.githubusercontent.com/121550022/219618724-d86a2d21-19f5-4e8d-8169-b8b38e4ec48a.png)

_webelement is undefined 
![image](https://user-images.githubusercontent.com/121550022/219617841-df3ede9c-0f3e-4851-b17d-fac68678c687.png)
Any further interaction with this control fails as there is no webelement set

**Expected behavior**
In order to use WDI5 with Chromium
Either:
WDI5 works completely without chromedriver
Or: 
At least works with chromium and chromedriver service combined

**Comment by dominikfeininger** [comment]
Do  other pure wdio tests which not use wdi5 like 
```js
it("should verify the browser title", async function () {
      const title = await browser.getTitle();
      expect(title).toEqual("xxx");
});
```

or 

```js
      const button = await this.browser.$('#some button id')
      await button.click()

```

work?

**Comment by QA-ler** [comment]
Yes, this works
see
![image](https://user-images.githubusercontent.com/121550022/219673762-c8703dc8-f6d4-40b6-854a-35367a966358.png)


Also getting Puppeteer or other expects on browser are working.

**Comment by vobu** [comment]
this relates to #404 and #168 - we seem to have a bug in wdi5 when run with the „plain“ devtools- instead of the webdriver-protocol. we‘ll try to tackle this as soon as time permits…but given that you‘re obviously really deep into debugging wdi5 already @QA-ler, any chance that you could go down the rabbit hole further and help hunt down the culprit 🐛 ?!? 😸

**Comment by QA-ler** [comment]
@vobu @dominikfeininger 
What i found out yet is that when chromedriver service is not used it uses the puppeteer evaluate function which behaves differently.
See
![image](https://user-images.githubusercontent.com/121550022/220404848-2798de2f-aaf2-439c-b91d-7c0679268a23.png)

Like mentioned in webdriver documentation https://webdriver.io/docs/api/browser/execute it is intended to return a webelement if possible. 
While puppeteer(devtools protocol) seems to have problems returning complex data in general.

Did not yet find a solution how to archive the same behaviour with devtools protocol.

Do one of you have an idea?

**Comment by QA-ler** [comment]
@dominikfeininger  @vobu 
I'v now tried to use browser.execute function directly in getControl function as fallback.
![image](https://user-images.githubusercontent.com/121550022/220628122-d29fffd1-478c-4304-9c8b-a4b422f0e083.png)

But as you see the element id looks completely different. 

At the moment further interactions still fail.
Maybe because of different element reference but I'm not sure about this.

Any ideas from your side?

**Comment by vobu** [comment]
`w00t`, thanks for digging deep and getting working code done!
The easiest way to directly contribute is to
- fork the project into your own gh space: https://docs.github.com/de/get-started/quickstart/fork-a-repo
- commit your code changes in a branch in your fork (and don't forget to push 😸 )
- head back to the original `wdi5` repo and propose a Pull Request (https://docs.github.com/de/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) with the branch in your fork as "source" and `wdi5`'s "main" branch as the "target"

Then CI runs, we can review the PR and work together on getting your changes in as fast as possible!

**Comment by QA-ler** [comment]
#430 
Done. Hope I did this right ^^

**Comment by vobu** [comment]
closed via #430
